### ISSUE TITLE
[BUG] - `api_get_specification` is broken

### ISSUE THREAD

**Issue by nkaretnikov** [conda-incubator/conda-store#761]
### Describe the bug

The `api_get_specification` HTTP endpoint seems to be completely broken. There are two bugs here:

Bug 1:
```py
            task, solve_id = conda_store.register_solve(db, specification)
            task.wait()
```

The task here can be a `str` sometimes, which causes a `ValueError`. I think it needs to be wrapped into `AsyncResult` here.

Bug 2:

```py
        return {"solve": solve.packages}
```

There's no `.packages` in `Solve` anymore. This was introduced when the DB schema was refactored, way before I was even part of the project. The new proper name is `package_builds`.

### Expected behavior

The endpoint `return`s some data instead of raising an error.

### How to Reproduce the problem?

Just call the endpoint. Some tests call it as well, which is how I noticed, but maybe those tests expect status codes 500 or 400, so they all pass. Those tests need to be looked at.

### Output

This is printed to the terminal:

```
'str' object has no attribute 'wait'
```

### Versions and dependencies used.

Current main (988445bfbc7de8e34ee36e2128471a9869c605cd).

**Comment by trallard** [comment]
@peytondmurray can you check if this is still an issue?

**Comment by peytondmurray** [comment]
A quick search of the tests doesn't show any that directly send a GET to `/api/v1/specification/`, every one of them is a POST. But sending a GET to `/api/v1/specification/` indeed raises an error, and I get a 500 response from the server:

![Image](https://github.com/user-attachments/assets/3a0c4825-1f32-4c37-b8b2-fa49d2a15a67)

**Comment by peytondmurray** [comment]
Closing in favor of #615, should be fixed by #913.
